### ISSUE TITLE
Pristine-1.4: Shim-eth asked to send PDU greater than MTU

### ISSUE THREAD

**Issue by kewinrausch** [IRATI/stack#841]
I got this error while testing on a real k=4 (36 nodes) DC fattree topology, sending PDU  with traffic generator of 1400 byte size (header not included).

```
Jan 26 10:24:34 a1 kernel: [62403.389998] rina-shim-eth-vlan(ERR): SDU too large (1738), dropping
Jan 26 10:24:34 a1 kernel: [62403.390040] rina-normal-ipc(ERR): Could not send to RMT (using port_id)
```

This is perfectly legal, but after this error I was no more able to run another test on that node (the client one), probably because the dropped PDU was a MGMT one. I said **probably** because I have trace of the type of dropped PDU and the behavior seems to be the one I got when dropping MGMT PDUs from RMT when the queue was full.

**Comment by kewinrausch** [comment]
After some test, with Eduard and Leonardo help, we identified this issue in the Management layer sending big packets because the number of nodes in the experiment (and so lot of data, not fragmented, was sent).

**Comment by edugrasa** [comment]
This issue should be solved by two developments: 1) use a reliable N-flow to transport N-layer management information between neighbor IPCPs ( #415) and implement delimiting to fragment SDUs that are larger than the N-1 layer SDU size ( #761).

It can also be mitigated by increasing the Ethernet MTU (up to 9000 bytes) or we could make the link-state routing policy aware of this issue and send flow state object messages that are smaller than a certain "safe" value (like 1300 bytes for instance, allowing to have several N-1 DIFs below).

**Comment by kewinrausch** [comment]
Ping... just wanted to remember you of this, but **not** because I want it done now, but because if someone want to use IRATI and have a more than 30 nodes DIF with the Eth-shim, then messages are dropped and the RIB/management layer does not work anymore (same error as RMT scheduling policy dropping a Management PDU, in the end). 

We discussed the out-of-sync problem that can arise between nodes, and this can be a trigger of it.

A generic solution can be having the EFCP implementing the fragmentation, so that every traffic which flows through a port is evaluated, fragmented and re-assembled in the destination, overcoming the problem. Still, if a fragment is lost and could not be recovered, the problem remains. But in the general case it should allow to have way bigger DIF. 

What do you think about this, @vmaffione? You seemed to have some additional comments for it during the meeting.

**Comment by vmaffione** [comment]
Yes, a way to mitigate the problem would be to limit the number of objects per CDAP message (typically M_WRITE), using more CDAP messages. The number of objects per message could be initially hardcoded (e.g. 10) and independent on the message type. During enrollment or routing updates it typically happens that a list of objects needs to be sent into a CDAP message: then we should split the list in sublists of limited size.

This is a clearly workaround but at least it would make it possible to scale a DIF to an arbitrary number of nodes.

We could the status of the current code to check that this is feature is actually feasible.

**Comment by edugrasa** [comment]
Problem is this can only be done before calling the CDAP API, because CDAP receives an object value that is already encoded. So it has to be done in the individual components/policies that call CDAP.. I've done in for the routing ps in #1024

**Comment by edugrasa** [comment]
Partial fix merged, real fix (implement delimiting) already in #761
